editoast/src/error.rs

```diff
@@ -170,3 +170,34 @@ impl EditoastError for reqwest::Error {
         "editoast:ReqwestError"
     }
 }
+
+#[derive(Debug)]
+pub enum EditoastUrlError {
+    Host,
+    Port,
+    Username,
+    Password,
+}
+
+impl Error for EditoastUrlError {}
+
+impl Display for EditoastUrlError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match *self {
+            EditoastUrlError::Host => write!(f, "Invalid host"),
+            EditoastUrlError::Port => write!(f, "Invalid port"),
+            EditoastUrlError::Username => write!(f, "Invalid username"),
+            EditoastUrlError::Password => write!(f, "Invalid password"),
+        }
+    }
+}
+
+impl EditoastError for EditoastUrlError {
+    fn get_status(&self) -> StatusCode {
+        StatusCode::INTERNAL_SERVER_ERROR
+    }
+
+    fn get_type(&self) -> &str {
+        "editoast:EditoastUrlError"
+    }
+}
```

Comment:
You should not implement `EditoastError` and `Error` manually.
You can use macros such as `thiserror::error` and `editoast_derive::editoast_error`.

Reply:
Done.